gobblin-runtime/src/main/java/org/apache/gobblin/runtime/TaskStateCollectorService.java

```diff
@@ -235,6 +255,49 @@ public Void call() throws Exception {
     this.eventBus.post(new NewTaskCompletionEvent(ImmutableList.copyOf(taskStateQueue)));
   }
 
+  /**
+   * Uses the size of work units to determine a job's progress and reports the progress as a percentage via
+   * GobblinTrackingEvents
+   * @param taskState of job launched
+   */
+  private void reportJobProgress(TaskState taskState) {
+    String stringSize = taskState.getProp(ServiceConfigKeys.WORK_UNIT_BYTE_SIZE);
+    if (stringSize == null) {
+      LOGGER.warn("Expected to report job progress but work unit byte size property null");
+      return;
+    }
+
+    Long taskByteSize = Long.parseLong(stringSize);
+    // if progress reporting is enabled, value should be present
+    String strTotalSizeToCopy = this.jobState.getProp(AbstractJobLauncher.TOTAL_BYTES_TO_COPY);
```

Comment:
You can use `PropertiesUtils.getPropAsLong` here (and for task byte size) instead of pulling the String. Instead of checking for null, check that the property exists in the state.